### ISSUE TITLE
LaTeX to markdown conversion: math mode next to digit converted incorrectly

### ISSUE THREAD

**Issue by abc-sophize** [jgm/pandoc#7058]
Pandoc Version: Online at pandoc.org/try [link](https://pandoc.org/try/?text=5%5C(-%5C)8+%5C(x%5C)&from=latex&to=markdown&standalone=0)

Input:
```
5\(-\)8 \(x\)
```

Actual Output (bug):
```
5$-$8 $x$
```

Converting this actual output back to LaTeX ([pandoc.org/try link](https://pandoc.org/try/?text=5%24-%248+%24x%24%0A&from=markdown&to=latex&standalone=0)), we get
```
5\$-\$8 \(x\)
```

Expected Output: 
Not quite sure.

Real world use case: conversion of [this TeX file](https://github.com/planetmath/26_Real_functions/blob/d43833d3f1a2b2f42b1833b4a703dc111ad32a9f/26A36-IntegrationOfDifferentialBinomial.tex) (line 47).

**Comment by jgm** [comment]
It's hard to get absolutely reliable round trips to and from markdown, because of issues like this.
I don't see a good way to avoid the problem.

**Comment by abchugh** [comment]
So there is no way to correctly represent the provided input in pandoc's markdown? Then, isn't that a bug in pandoc's spec?

If fixing the spec is out-of-scope, here's a workaround: Add a space after MathInline sections that are immediately followed by a digit. Without this, the output is truly awful. Eg the paragraph on Iine 47 of [this TeX file](https://github.com/planetmath/26_Real_functions/blob/d43833d3f1a2b2f42b1833b4a703dc111ad32a9f/26A36-IntegrationOfDifferentialBinomial.tex) is completely ruined.

In any case, thank you for your consideration of my request and building such a fantastic library. With pandoc's help I have been able to put 1000's of latex docs on the web which wouldn't have been possible otherwise.

**Comment by jgm** [comment]
> pandoc's spec?

We don't really have a spec for pandoc's markdown.  (There is one for commonmark, but it doesn't support math.)

I think one possible, albeit awkward workaround would be to emit

```
5$-$<!-- -->8 $x$
```
The HTML comments would separate `5` and `$` so that pandoc would know it's safe to interpret as math.

The markdown writer would have to check for math inlines that appear immediately before digits, and insert this filler.

**Comment by jgm** [comment]
Or zero-width space

```
5$-$&#x200B;8
```
Unlike the HTML comment, this would cause an additional character to be emitted in many formats, but it's zero-width.

**Comment by jgm** [comment]
I'll go with the HTML comment version; this kind of thing should be fairly uncommon anyway.

**Comment by jgm** [comment]
closed by 9223788a05fe619d567bcbdf4cb31db63de86f32

**Comment by abchugh** [comment]
Thanks, John.
